jwst/regtest/test_miri_mrs_badpix_selfcal.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from astropy.io.fits.diff import FITSDiff
+
+from jwst.badpix_selfcal import BadpixSelfcalStep
+
+
+@pytest.fixture(scope="module")
+def run_pipeline(rtdata_module):
+
+    rtdata = rtdata_module
+    rtdata.get_asn("miri/mrs/jw01204-o021_20240127t024203_spec2_00010_asn.json")
+    BadpixSelfcalStep.call(rtdata.input, skip=False, save_results=True, flagfrac=0.0005)
```

Comment:
Might also be a good idea to add a regtest for running it from the spec2 pipeline, with skip=False.